target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-rax120.dts

```diff
@@ -0,0 +1,1010 @@
+		    qcom,num-queue = <4>;
+		    qcom,num-irq = <9>;
+		    qcom,num-pri = <4>;
+		    qcom,load-addr = <0x40000000>;
+		    qcom,low-frequency = <748800000>;
+		    qcom,mid-frequency = <1497600000>;
+		    qcom,max-frequency = <1689600000>;
+		    qcom,bridge-enabled;
+		    qcom,ipv4-enabled;
+		    qcom,ipv4-reasm-enabled;
+		    qcom,ipv6-enabled;
+		    qcom,ipv6-reasm-enabled;
+		    qcom,wlanredirect-enabled;
+		    qcom,tun6rd-enabled;
+		    qcom,l2tpv2-enabled;
+		    qcom,gre-enabled;
+		    qcom,gre-redir-enabled;
+		    qcom,map-t-enabled;
+		    qcom,portid-enabled;
+		    qcom,ppe-enabled;
+		    qcom,pppoe-enabled;
+		    qcom,pptp-enabled;
+		    qcom,tunipip6-enabled;
+		    qcom,shaping-enabled;
+		    qcom,wlan-dataplane-offload-enabled;
+		    qcom,vlan-enabled;
+		    npu-supply = <&npu_vreg>;
+	};
+
+	usbbam@8B04000 {
+		compatible = "qcom,usb-bam-msm";
+		reg = <0x8B04000 0x17000>;
+		interrupt-parent = <&intc>;
+		interrupts = <GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>;
+
+		qcom,bam-type = <0>;
+		qcom,usb-bam-fifo-baseaddr = <0x4A600000>;
+		qcom,usb-bam-num-pipes = <4>;
+		qcom,ignore-core-reset-ack;
+		qcom,disable-clk-gating;
+		qcom,usb-bam-override-threshold = <0x4001>;
+		qcom,usb-bam-max-mbps-highspeed = <400>;
+		qcom,usb-bam-max-mbps-superspeed = <3600>;
+		qcom,reset-bam-on-connect;
+
+		qcom,pipe0 {
+			label = "ssusb-qdss-in-0";
+			qcom,usb-bam-mem-type = <2>;
+			qcom,dir = <1>;
+			qcom,pipe-num = <0>;
+			qcom,peer-bam = <0>;
+			qcom,peer-bam-physical-address = <0x6044000>;
+			qcom,src-bam-pipe-index = <0>;
+			qcom,dst-bam-pipe-index = <0>;
+			qcom,data-fifo-offset = <0x0>;
+			qcom,data-fifo-size = <0xe00>;
+			qcom,descriptor-fifo-offset = <0xe00>;
+			qcom,descriptor-fifo-size = <0x200>;
+		};
+	};	
+
+	dbm_1p5: dbm@8AF8000 {
+		compatible = "qcom,usb-dbm-1p5";
+		reg = <0x8AF8000 0x300>;
+		qcom,reset-ep-after-lpm-resume;
+	};
+};
+
+&blsp1_i2c2 {
+	status = "okay";
+	fan_left: g761@3e {
+		compatible = "gmt,g761";
+		reg = <0x3e>;
+		clocks =<&g761_clk>;
+		fan_gear_mode = <0>;
+		fan_starty = <1>;
+		pwm_polarity = <0>;
+	};
+};
+
+&mdio {
+	status = "okay";
+	pinctrl-0 = <&mdio_pins>;
+	pinctrl-names = "default";
+	reset-gpios = <&tlmm 37 GPIO_ACTIVE_LOW>;
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+	phy2: ethernet-phy@2 {
+		reg = <2>;
+	};
+	phy3: ethernet-phy@3 {
+		reg = <3>;
+	};
+	phy4: ethernet-phy@4 {
+		reg = <4>;
+	};
+	phy5: ethernet-phy@5 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <7>;
+	};
+};
+
+&blsp1_uart3 {
+	status = "okay";
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&pcie0 {
+	perst-gpio = <&tlmm 58 1>;
+};
+
+&pcie1 {
+	perst-gpio = <&tlmm 61 1>;
+};
+
+&qpic_bam {
+	status = "okay";
+};
+
+&qpic_nand {
+	status = "okay";
+	nand@0 {		
+		reg = <0>;
+		#address-cells = <0x01>;
+		#size-cells = <0x01>;
+		nand-ecc-strength = <4>;
+		nand-ecc-step-size = <512>;
+		nand-bus-width = <8>;
+		partition@0 {
```

Comment:
Cant, you just populate it using smem instead of doing it manually?

Reply:
The partition info? smem provides the partition layout represented by QCA however this is incomplete. Netgear overrides it by hard coding their partition layout in ofpart.c that was used as a foundation for these entries.

Reply:
Ahh, so they are still pulling that crap by hardcoding the partitions in the kernel source.

Reply:
Unfortunately. I was actually hoping to define only the variations identified beyond smem but I found that it skips the smem import with this node present so we have some redundancy I suppose but it's consistent at least.

Reply:
No, you can only have one parser at a time.
By priority, cmdline the ofparts and only them SMEM currently.